app/models/course.rb

```diff
@@ -149,6 +149,27 @@ def application_forms
       .distinct
   end
 
+  def self.find_courses_in_previous_cycle(provider)
+    previous_cycle
+        .joins(:course_options)
+        .distinct
+        .where(provider: provider)
+        .includes(%i[provider accredited_provider])
+  end
+
+  def self.find_courses_in_current_cycle(provider)
```

Comment:
This method is identical to the `find_` method above, apart from the base scope, I think we can reduce this.
There's already a `in_cycle` scope defined in this class. We could call this.

Reply:
I've added a new scope and removed these methods